src/ring/middleware/json.clj

```diff
@@ -11,7 +12,10 @@
 (defn- read-json [request & [{:keys [keywords? bigdecimals?]}]]
   (if (json-request? request)
     (if-let [body (:body request)]
-      (let [body-string (slurp body)]
+      (let [encoding (or (:character-encoding request)
```

Comment:
The `:character-encoding` key is deprecated, and should be identical to the output of `character-encoding`. So we can just write:

```clojure
(or (character-encoding request) "UTF-8")
```